packages/ra-core/src/dataProvider/withLifecycleCallbacks.ts

```diff
@@ -332,7 +332,7 @@ export const withLifecycleCallbacks = (
 
         create: async function <RecordType extends RaRecord = any>(
             resource: string,
-            params: CreateParams<RecordType>
+            params: CreateParams<Partial<RecordType>>
```

Comment:
I don't think this should be changed - to me, it's the CreateParams and UpdatePArams that should be changed. 

```diff
export interface CreateParams<T = any> {
-   data: T;
+   data: Partial<T>;
    meta?: any;
}
```

Reply:
Agreed and done